Implement some of the conversion opcodes

Leonardo2718 opened the pull request:
This changeset implements several of the WebAssembly type conversion opcodes.

Closes #79 
Closes #80 
Closes #81 
Closes #82 
Closes #83
Closes #86

aviansie-ben commented on src/jit/function-builder.cc:
```diff
@@ -693,6 +693,76 @@ bool FunctionBuilder::Emit(TR::BytecodeBuilder* b,
+      Push(b,
+           "i32",
+      b->  UnsignedConvertTo(Int32,
+      b->            LoadAt(typeDictionary()->PointerTo(Int8), EmitMemoryPreAccess<int8_t>(b, &pc))));
```
Just a small nitpick, but the indentation is wrong on all your `UnsignedConvertTo`s for unsigned extending loads

Leonardo2718 commented:
@aviansie-ben I've corrected the indentation, pushed some additional opcode implementations, and removed the WIP.

Although there are no tests for these opcodes, we should go ahead and merge this PR so other work can move forward, which will act as testing for these opcodes.
